kedro/ipython/__init__.py

```diff
@@ -178,3 +184,124 @@ def _find_kedro_project(current_dir: Path) -> Any:  # pragma: no cover
         current_dir = current_dir.parent
 
     return None
+
+
+@typing.no_type_check
+@magic_arguments()
+@argument(
+    "node",
+    type=str,
+    help=("Name of the Node."),
+    nargs="?",
+    default=None,
+)
+def magic_load_node(node: str) -> None:
+    """The line magic %load_node <node_name>
+    Currently it only support Jupyter Notebook (>7.0) and Jupyter Lab. This line magic
+    will generate code in multiple cells to load dataest from Data Catalog, import
+    relevant functions and modules, and the function body.
+    """
+    cells = _load_node(node)
+    from ipylab import JupyterFrontEnd
```

Comment:
Trying this on IPython (after naming my node, see above) and got an error:

```
ModuleNotFoundError: No module named 'ipylab'
```

I would find it confusing if `kedro.ipython` didn't work on IPython, just because of the name. On the other hand, I don't think it makes sense for this to be in IPython at all, because the point is to bring the source code and be able to edit it on a cell - something that can't happen on a REPL.

If this is to be a Jupyter-only extension, I think we should have it somewhere else. Maybe `kedro.jupyter`? Or even a separate `kedro-jupyter` package?

Reply:
Sorry, there are two different things here:

- I did not reinstall kedro so I didn't have `ipylab`, but I see it's there when the user does `pip install kedro[jupyter]` 👍🏽 
- After installing it, I did `%load_node clean_statuses` on IPython and nothing happened:

<img width="891" alt="image" src="https://github.com/kedro-org/kedro/assets/316517/d8429f33-14f7-4678-876a-734c96763a77">

My node function is not in `dir()`:

```
In [4]: "clean_statuses" in dir()
Out[4]: False
```

so I stand by my previous point: since this is pointless in IPython, I'd move it somewhere else.

Reply:
https://github.com/kedro-org/kedro/issues/3536

There is a separate ticket for IPython support. Originally I thought the same, but it appears that some people, particular anti-notebook user would still like to have IPython support. It will work slightly different because IPython doesn't have the concept of "cells". It will still be useful if data can get pre-load (or just the code to load the corresponding data from catalog). If you have comments about the design of the feature, feel free to drop comments over there instead.

Re: `kedro.ipython` vs `kedro.jupyter` - I agree it's a bit confusing, but I think it's okay to leave it in `kedro.ipython` since ultimately I plan to add support for IPython. In additionally, we almost use IPython/Jupyter interchangeably because the feature parity is always the same, until now.

Reply:
Sounds good. I don't know how feasible this is, but if it's possible to show a warning like "IPython is not yet supported", that would be golden.